Mathlib/Algebra/Algebra/Quasispectrum.lean

```diff
@@ -0,0 +1,291 @@
+/-
+Copyright (c) 2024 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import Mathlib.Algebra.Algebra.Unitization
+import Mathlib.Algebra.Algebra.Spectrum
+
+/-!
+# Quasiregularity and quasispectrum
+
+For a non-unital ring `R`, an element `r : R` is *quasiregular* if it is invertible in the monoid
+`(R, ∘)` where `x ∘ y := y + x + x * y` with identity `0 : R`. We implement this both as a type
```

Comment:
Does this weird multiplication have a name? I find it a bit disturbing that `Quasiregular R` isn't actually the type of quasiregular elements of `R`. I can't really think of a better name for the structure though.

Reply:
This bothered me as well, but I also couldn't think of a better name. I thought about `Quasi` but that seemed *far* too generic. I can't find a name for this multiplication anywhere. The text from which I garnered some of this was "Exercises in classical ring theory" by T.Y. Lam, but the author doesn't name this operation. It also doesn't appear to have a name in the older work "Real banach algebras" by Lars Ingelstam. So : shrug:

Reply:
Maybe something like `PreQuasiregular R`, or `ExtQuasiregular R`...?